app/controllers/assignment_swap_controller.rb

```diff
@@ -0,0 +1,34 @@
+class AssignmentSwapController < SecureController
+
+  skip_before_action :authenticate_user!
+
+  def index
+    @weeks = current_office.weeks.upcoming 
+  end
+
+  def create
+    #no assignments were selected
+    if !(params.has_key?("assignment"))
+      redirect_to office_assignment_swap_index_path
+      return
+    end
+
+    # only 1 or more than 2 assignments were selected
+    if params[:assignments][:assignment_id].count != 2
+      redirect_to office_assignment_swap_index_path
+      return
+    end
+
+    ass1 = Assignment.find(params[:assignments][:assignment_id].first)
+    ass2 = Assignment.find(params[:assignments][:assignment_id].second)
+
+    employee1 = ass1.employee
+    employee2 = ass2.employee
+
+    ass1.update(employee: employee2)
+    ass2.update(employee: employee1)
+
+    redirect_to office_assignment_swap_index_path
+  end
```

Comment:
Let's clean this method up a bit. Put comments directly into the code as extracted method name(s). Remove complex procedures to services. Combine duplicated responses. Now it's two readable lines! Nice!

```
SwapAssignments.call(assignment_ids: params[:assignments][:assignment_id]) unless less_than_two_assignments_selected?

redirect_to(action: :index)
```